### DISCUSSION TITLE
How to specify extra index or primary index url in pyproject.toml

### DISCUSSION THREAD

**Discussion by vit1251** [pypa/setuptools#3860] [Q&A]
How to specify primary Python Index in pyproject.toml with private repository wihout setup .pypirc and other user environment changes?

**Comment by abravalheri** [comment]
Hi @vit1251, `setuptools` no longer supports indexes.

You can try using a [PEP 440 direct reference](https://peps.python.org/pep-0440/#direct-references) when specifying your requirements, but please be aware that PyPI will not accept packages that rely on such configuration. Also note that this will depend on the installer (e.g. `pip`) supporting the specific protocol.
